src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1797,10 +1798,29 @@ applyTransaction (OpenView& view,
 
     try
     {
+        // If the txn signature has already been checked,
+        // then we'll get a result from the cache. If not,
+        // it'll check now (this can only happen in the case
+        // where we first see the txn during consensus,
+        // OR the cache expires). Exception: If the txn
+        // does not have an sfAccount field, skip the check.
+        // The only transactions that do not need sfAccount,
+        // are also the only transactions that don't get
+        // signed. Any others will be rejected as malformed.
+        if (txn->isFieldPresent(sfAccount) &&
+            getApp().getHashRouter().checkValidity(*txn,
+#if RIPPLE_ENABLE_MULTI_SIGN
```

Comment:
Since `Account` is `SOE_REQUIRED` in every transaction, the `isFieldPresent` test doesn't do anything. We need to make sure this code doesn't reject pseudo-transactions.